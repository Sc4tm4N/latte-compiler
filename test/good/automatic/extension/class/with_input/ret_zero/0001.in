1
klapek
2
luzno
10
szura
