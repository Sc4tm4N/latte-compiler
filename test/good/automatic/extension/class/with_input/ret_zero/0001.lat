int main () {
  Tester t = new Tester;
  t.s = "";

  t.accumulateI();
  t.accumulateS();

  printInt(t.i);
  printString(t.s);

  t.accumulateI();
  t.accumulateS();

  printInt(t.i);
  printString(t.s);

  t.accumulateI();
  t.accumulateS();

  printInt(t.i);
  printString(t.s);

  return 0;
}

class Tester {
  int i;
  string s;

  void accumulateI() {
    i = i + readInt();
  }

  void accumulateS() {
    s = s + readString();
  }

}
