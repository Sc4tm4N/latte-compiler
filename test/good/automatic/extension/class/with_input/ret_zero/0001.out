1
klapek
3
klapekluzno
13
klapekluznoszura
